elasticapm/traces.py

```diff
@@ -198,8 +261,12 @@ def to_dict(self):
             result["context"] = self.context
         return result
 
+    def track_span_duration(self, span_type, span_subtype, self_duration):
+        with self._duration_lock:
+            self._span_timers[(span_type, span_subtype)].update(self_duration)
```

Comment:
perhaps return early if the transaction is already ended?

Reply:
we currently don't support asynchronous spans, so this shouldn't happen. But I'll add a comment to indicate that this should be checked once we have asynchronous spans.